test/phabricator_tests/test_phabricator.py

```diff
@@ -0,0 +1,658 @@
+        """
+        Tests generate_phids function (successful, user_query_usernames does not fail)
+        """
+        expected_list_of_phids, expected_raw_response = self.mock.generate_phids_response()
+        mock_user_query_usernames.return_value = self.mock.user_query_usernames()
+
+        list_of_phids, raw_response = PhabricatorService().generate_phids(
+            "user_names",
+            None)
+
+        self.assertEqual(expected_list_of_phids, list_of_phids)
+        self.assertEqual(expected_raw_response, raw_response)
+        mock_user_query_usernames.assert_called_with(
+            "user_names",
+            None
+        )
+
+    @patch(PATH + 'PhabricatorService.user_query_usernames',
+           side_effect=InternetError)
+    def test_generate_phids_failure(self, mock_user_query_usernames):
+        """
+        Tests generate_phids function (successful, user_query_usernames does fail)
+        """
+
+        with self.assertRaises(InternetError):
+            list_of_phids, raw_response = PhabricatorService().generate_phids(
+                "user_names",
+                None)
+
+        mock_user_query_usernames.assert_called_with(
+            "user_names",
+            None
+        )
+
+    @patch(PATH + 'PhabricatorService.get_comments')
+    @patch(PATH + 'PhabricatorService.get_last_comment')
+    @patch(PATH + 'PhabricatorService.time_from_epoch')
+    @patch(PATH + 'PhabricatorService.check_request_state')
+    @patch(PATH + 'PhabricatorService.has_new_comments')
+    @patch(PATH + 'PhabricatorService.author_data')
+    @patch(PATH + 'PhabricatorReview')
+    def test_get_reviews_successful_no_last_comment(self,mock_PhabricatorReview,
+                                                    mock_author_data,
+                                                    mock_has_new_comments,
+                                                    mock_check_request_state,
+                                                    mock_time_from_epoch,
+                                                    mock_get_last_comment,
+                                                    mock_get_comments):
+        """
+        Tests get_reviews function with simple parameters (i.e. no show_last_comment, duration, value etc.)
+        """
+
+        mock_PhabricatorReview.return_value = MagicMock()
+        mock_time_from_epoch.return_value = "mock_date"
+        fake_phab = MagicMock()
+        fake_reviews = self.mock.get_reviews()
+        mock_get_comments.return_value = "test comment"
+        mock_check_request_state.return_value = True
+        mock_author_data.return_value = ({'userName': 'mock_user', 'image': 'mock.com'}, [])
+
+        response = PhabricatorService().get_reviews(
+            phab=fake_phab,
+            reviews=fake_reviews,
+            raw_response=[],
+            host="www.google.com",
+            state_=None,
+            value=None,
+            duration=None,
+            show_last_comment=None)
```

Comment:
Same as above, i think you can come up with something to reduce it. Maybe it would be beneficial to define global dict in new class and then use it to call the method.

```python
class TestPhabricatorGetReviews(TestCase):
    self.setUp(self):
        # initializing fake_phab and fake_reviews goes here
        DEFAULT_PARAMS = {
            'phab': fake_phab,
            'reviews': fake_reviews,
            'host': 'www.google.com',
            ...
        }

    def test_something(self):
        # editing DEFAULT_PARAMS per each test to correspond to specified test goes here
        response = PhabricatorService().get_reviews(**self.DEFAULT_PARAMS)
```